eradiate/scenes/core.py

```diff
@@ -79,6 +80,13 @@ def __iter__(self):
         return self.data.__iter__()
 
     def __setitem__(self, k, v):
+        try:
+            self.data.__getitem__(k)
+            warnings.warn(
+                f"Duplicate scene element id '{k}' found. Scene elements will be overwritten."
+            )
+        except KeyError:
+            pass
```

Comment:
I think a simple presence check would be more legible. Also, you're not setting scene element IDs here, so I'd make the message more general:
```suggestion
        if k in self.data:
            warnings.warn(f"Duplicate key '{k}' will be overwritten.")
        return self.data.__setitem__(k, v)
```

Reply:
Well but the second message does not tell the user anything about where this issue comes from. If you don't know how the `KernelDict` works, you have no idea how to fix this. :(

Reply:
Okay, so how about something like this:
```
Duplicate key '{k}' will be overwritten. Are you trying to add scene elements with duplicate IDs to this KernelDict?
```
This points the user to the potential issue without being definitive.

Reply:
That's nice :)